exec/exec.go

```diff
@@ -0,0 +1,29 @@
+// Package exec adapts the lego exec DNS
+// provider for Caddy. Importing this package plugs it in.
+package exec
```

Comment:
Can we name this package something else? It conflicts with the standard library `exec` package.

Reply:
I found it confusing at first, too, but then I found it *more* confusing if it were the only lego wrapper that is called different than the package that it wraps.
You don't mean a *technical* conflict, do you? The only conflict could happen in caddy/caddymain/run.go where it's plugged in and they already get plugged in with an underscore, so there shouldn't be any conflict.
So it's just about humans... would "command" be any better? "generic", "script"?

Reply:
I like `exec` for the fact that it's the same as in the lego repository. But `generic` is fine too, as it seems we can both live with that.